pkgs/tools/filesystems/bees/bees-service-wrapper

```diff
@@ -0,0 +1,108 @@
+#!@bash@/bin/bash
+PATH=@coreutils@/bin:@utillinux@/bin:@btrfsProgs@/bin:$PATH
+beesd_bin=@bees@/lib/bees/bees
+# PLEASE KEEP NIX-ISMS ABOVE THIS LINE TO EASE UPSTREAM MERGE
+#!/usr/bin/env bash
+
+# Upstream wrapper requires UUID to be used for configuration.
+
+# However, when declaratively describing a host, we may not know its UUID, and
+# shouldn't need to persist something that will differ between hosts built from
+# the same configuration template.
+
+# Thus, for using bees from NixOS, we have our own wrapper, which supports not
+# just UUID but any specification permitted by findmnt
+
+[[ $bees_debug ]] && { PS4=':${BASH_SOURCE##*/}:$LINENO+'; set -x; }
+
+usage() {
+  cat >&2 <<EOF
+Usage: ${BASH_SOURCE##*/} run|cleanup config-name|fs-spec[;idxSizeMB=...][;verbosity=...][;workDir=...]
+
+  fs-spec should be in a format recognized by find-mnt.
+EOF
+  exit 1
+}
+
+die() { echo "$*" >&2; exit 1; }
+
+do_setup() {
+  declare -n options_src=$1; shift
+  declare fs_spec=$1; shift
+  declare -g uuid bees_{work,run,mnt}_dir bees_verbosity bees_db_size{,_mb} blockdev
+  for key in "${!options_src[@]}"; do
+    case $key in
+      workDir) bees_work_dir=${options_src[$key]} ;;
+      verbosity) bees_verbosity=${options_src[$key]}  ;;
+      idxSizeMB) bees_db_size_mb=${options_src[$key]} ;;
+    esac
+  done
+
+  : bees_config_dir="${bees_config_dir:=/etc/bees}"
+  if [[ -s $bees_config_dir/$fs_spec.conf ]]; then
+    source "$bees_config_dir/$fs_spec.conf"
+  fi
+  read -r uuid fstype < <(findmnt -n -o uuid,fstype "$fs_spec") && [[ $fstype ]] || exit
+  for f in "${bees_config_dir}/$fs_spec.conf" "$bees_config_dir/UUID=$uuid.conf"; do
+    [[ -s $f ]] && source "$f"
+  done
+  : bees_work_dir="${bees_work_dir:=.beeshome}"
+  : bees_run_dir="${bees_run_dir:=/run/bees}"
+  : bees_mnt_dir="${bees_mnt_dir:=$bees_run_dir/mnt/$uuid}"
+  : bees_home="${bees_home:=$bees_mnt_dir/$bees_work_dir}"
+  : bees_verbosity="${bees_verbosity:-6}"
+  : bees_db_size_mb="${bees_db_size_mb:=1024}"
+  : blockdev=${blockdev:="/dev/disk/by-uuid/$uuid"}
+  bees_db_size="$(( bees_db_size_mb * 1024 * 1024 ))"
+
+  [[ -b $blockdev ]] || die "Block device $blockdev missing"
+  [[ $fstype = btrfs ]] || die "Device type is $fstype, not btrfs"
+  (( bees_db_size_mb % 16 == 0 )) || die "DB size must be divisible by 16MB"
+}
+
+do_run() {
+  local db old_db_size
+
+  do_setup "$@"
+  mkdir -p -- "$bees_mnt_dir" || exit
+  mountpoint -q "$bees_mnt_dir" || mount -osubvolid=5 -- "$blockdev" "$bees_mnt_dir" || exit
```

Comment:
Does `subvolid` equal to 5 have some special meaning? Maybe a comment should explain it then…

Reply:
Yes, 5 is special -- it refers to the btrfs root subvolume. Agreed that a comment is appropriate.